### PR TITLE
DRYD-1172: Rename supplied birth date fields

### PR DESCRIPTION
**Pull request by mikejritter**
Jira: https://collectionspace.atlassian.net/browse/DRYD-1172

* Rename supplied birthDate fields to avoid conflicts with previously defined fields
* Update messages for supplied birth date fields better indicate what they are in search
* Add groupName message for birth date structured date input
* Fix import of structured date fields for suppliedStructuredBirthDate
* Add missing dataType definitions
* Remove errant repeat declarations for suppliedStructuredBirthDate and suppliedBirthPlace

### CODE REVIEW

File: src/plugins/recordTypes/person/fields.js [hunk 1]
```diff
@@ -1259,31 +1270,35 @@ export default (configContext) => {
...
-            suppliedBirthDate: {
+            suppliedStructuredBirthDateGroup: {
               [config]: {
+                dataType: DATA_TYPE_STRUCTURED_DATE,
                 messages: defineMessages({
                   fullName: {
-                    id: 'field.persons_common.suppliedBirthDate.fullName',
-                    defaultMessage: 'Birth date supplied',
+                    id: 'field.persons_common.suppliedStructuredBirthDateGroup.fullName',
+                    defaultMessage: 'Supplied structured birth date',
```
Comment (ray-lee) [inline src/plugins/recordTypes/person/fields.js:1279]: We don't normally put "structured" in the label for structured dates. "Supplied birth date" should suffice.

File: src/plugins/recordTypes/person/fields.js [hunk 2]
```diff
@@ -1259,31 +1270,35 @@ export default (configContext) => {
...
+                  },
+                  groupName: {
+                    id: 'field.persons_common.suppliedStructuredBirthDateGroup.groupName',
+                    defaultMessage: 'Date',
```
Comment (ray-lee) [inline src/plugins/recordTypes/person/fields.js:1283]: groupName should in most cases be the same as name, so probably want "Supplied" here to match the name.

File: src/plugins/recordTypes/person/fields.js [hunk 3]
```diff
@@ -1259,31 +1270,35 @@ export default (configContext) => {
...
                     defaultMessage: 'Supplied',
                   },
                 }),
-                repeating: true,
                 view: {
                   type: StructuredDateInput,
                 },
-                ...extensions.structuredDate.fields,
               },
+              ...extensions.structuredDate.fields,
```
Comment (ray-lee) [inline src/plugins/recordTypes/person/fields.js:1294]: I wouldn't be surprised if this is the only reason the field didn't save, but the other changes are good to have too.

File: src/plugins/recordTypes/person/fields.js [hunk 4]
```diff
@@ -1223,17 +1229,21 @@ export default (configContext) => {
...
               type: CompoundInput,
             },
           },
-          birthDateGroup: {
+          suppliedBirthDateGroup: {
             [config]: {
               messages: defineMessages({
+                fullName: {
+                  id: 'field.persons_common.suppliedBirthDateGroup.fullName',
+                  defaultMessage: 'Supplied birth date',
```
Comment (ray-lee) [inline src/plugins/recordTypes/person/fields.js:1243]: This seems inconsistent with all the other supplied fields, e.g. "Gender supplied", "Pronoun supplied". Seems like this should be "Birth date supplied", or the others should be "Supplied gender," etc.

File: src/plugins/recordTypes/person/fields.js (hunk 1 above)
Comment (mikejritter) [reply src/plugins/recordTypes/person/fields.js:1279]: I was trying to find a way to differentiate between the `suppliedBirthDateGroup` and the `suppliedStructuredBirthDateGroup` in the search with this (previously they both had shared `Birth date supplied`). Maybe the message for the group should be different in that case? I'll look around to see if there are any fields similar to this to model after.

File: src/plugins/recordTypes/person/fields.js (hunk 4 above)
Comment (mikejritter) [reply src/plugins/recordTypes/person/fields.js:1243]: I had wanted these fields to be easier to find in the search (similar to the above issue, but they were all together under the birth date fields), so I'll try updating the others to be `Supplied ...` and see how that feels in the advanced search.

File: src/plugins/recordTypes/person/fields.js (hunk 1 above)
Comment (mikejritter) [reply src/plugins/recordTypes/person/fields.js:1279]: @ray-lee thoughts on adding a fullName to the groups for maker supplied info that would read `Maker supplied birth date`, `Maker supplied birth place`, etc. It's really only under adding the group conditions for searches but seems to make it a bit clearer imo.

File: src/plugins/recordTypes/person/fields.js (hunk 4 above)
Comment (mikejritter) [reply src/plugins/recordTypes/person/fields.js:1243]: I ended up going with Birth date supplied, and added on `supplied` to other fields to better group them together e.g. `Birth date supplied use restriction`, `Ethnicity supplied use restriction`. Felt like it made them a bit easier to find when looking for them in the search.

**Review by ray-lee — APPROVED** [review]
Ok, I think that's reasonable. We can see what people think. Naming is hard.

**Pull request merged**